inc/classes/optimization/class-remove-query-string.php

```diff
@@ -87,76 +86,79 @@ public function __construct( Options $options, $busting_path, $busting_url ) {
 	}
 
 	/**
-	 * Set DOM crawler from provided HTML content
+	 * Remove query strings for CSS files that have one
 	 *
 	 * @since 3.1
 	 * @author Remy Perona
 	 *
-	 * @param HtmlPageCrawler $crawler Crawler instance.
-	 * @param string          $html   HTML content.
-	 * @return void
+	 * @param string $html HTML content.
+	 * @return string
 	 */
-	public function set_crawler( HtmlPageCrawler $crawler, $html ) {
-		$this->crawler = $crawler::create( $html );
+	public function remove_query_strings_css( $html ) {
+		$styles = $this->find( '<link\s+([^>]+[\s\'\"])?href\s*=\s*[\'\"]\s*?([^\'\"]+\.css(?:\?[^\'\"]*)?)\s*?[\'\"]([^>]+)?\/?>', $html );
+
+		if ( ! $styles ) {
+			return $html;
+		}
+
+		foreach ( $styles as $style ) {
+			$url = $style[2];
+
+			$url = $this->can_replace( $url );
+
+			if ( ! $url ) {
+				continue;
+			}
+
+			$optimized_url = $this->replace_url( $url, 'css' );
+
+			if ( ! $optimized_url ) {
+				continue;
+			}
+
+			$replace_style = str_replace( $style[2], $optimized_url, $style[0] );
+			$html          = str_replace( $style[0], $replace_style, $html );
+		}
+
+		return $html;
 	}
 
 	/**
-	 * Remove query strings for CSS/JS files that have one
+	 * Remove query strings for JS files that have one
 	 *
 	 * @since 3.1
 	 * @author Remy Perona
 	 *
+	 * @param string $html HTML content.
 	 * @return string
 	 */
-	public function optimize() {
-		$style_nodes  = $this->find( 'link[href*=".css"]' );
-		$script_nodes = $this->find( 'script[src*=".js"]' );
+	public function remove_query_strings_js( $html ) {
+		$scripts = $this->find( '<script\s+([^>]+[\s\'\"])?src\s*=\s*[\'\"]\s*?([^\'\"]+\.js(?:\?[^\'\"]*)?)\s*?[\'\"]([^>]+)?\/?>', $html );
```

Comment:
No need to escape double quotes: `[\s\'\"]`, `[\'\"]`, `[^\'\"]` => `[\s"\']`, `["\']`, `[^"\']`.